lib/Jijo/Validation.hs

```diff
@@ -174,5 +178,19 @@ jValidateOptField fieldName vField o =
     mapJValidationReport (scopeJValidationReport (JPSField fieldName)) $
     vField field
 
+jValidateElements ::
+  (j -> JValidation e a) ->
+  Vector.Vector j ->
+  JValidation e (Vector.Vector a)
+jValidateElements vElement =
+  itraverse $ \i element ->
+    mapJValidationReport (scopeJValidationReport (JPSIndex i)) $
+    vElement element
+
 mapJValidationError :: (e -> e') -> JValidation e a -> JValidation e' a
 mapJValidationError f = mapJValidationReport (fmap f)
+
+-- Indexed 'traverse' using 'State'
```

Comment:
Is it different from https://hackage.haskell.org/package/vector-0.12.0.3/docs/Data-Vector.html#v:imapM?

Reply:
Yes, this doesn't require `Monad` which `JValidation` does not implement.

Reply:
I added a comment to clarify this.

Reply:
https://github.com/haskell/vector/issues/132 – sigh.